freqtrade/persistence/models.py

```diff
@@ -866,9 +866,17 @@ def calc_profit_ratio(self, rate: Optional[float] = None,
         return float(f"{profit_ratio:.8f}")
 
     def recalc_trade_from_orders(self):
+        filled_orders_count = len(self.select_filled_orders(self.entry_side))
+        latest_order_in_trade = self.select_order(self.entry_side, True)
+        # No fills but newer order
+        if (filled_orders_count == 0 and latest_order_in_trade is not None and
+                latest_order_in_trade.id is not None):
+            # after ensuring there is a populated order
+            if latest_order_in_trade.id > 1:
```

Comment:
Whatever you do here (or why) - it's wrong.

`.id` is an irrelevant element of the order (it's the sequence of the table) - and it does NOT correspond to the order ID for this trade - but simply the sequence of all orders (so a trade could have order.id 200 - and it's still the only order for this trade).

It'll be 1 for the very first trade (and in tests that only provide one order) - but nothing else.

what you need to check here instead is that `.price` is populated (that can be empty)

Reply:
Thanks for the feedback. Updated once more through: https://github.com/freqtrade/freqtrade/pull/6692/commits/ae01afdd0f3bd9f63b3b39d8bb22ae8a3a38830b
Hope I've understood your intent correctly.